src/pocketmine/entity/Monster.php

```diff
@@ -20,8 +20,200 @@
 */
 
 namespace pocketmine\entity;
-
+use pocketmine\event\entity\EntityDamageByEntityEvent;
+use pocketmine\event\entity\EntityDamageEvent;
+use pocketmine\Player;
+use pocketmine\level\Level;
+use pocketmine\entity\PositionAndOrientation;
+use pocketmine\entity\MobsControl;
+use pocketmine\network\protocol\AddEntityPacket;
+use pocketmine\item\Item as ItemItem;
 
 abstract class Monster extends Creature{
 
+	public $width = 0.6;
+	public $length = 0.6;
+	public $height = 1.8;
+	
+	protected $withinProximity = 10;
+	protected $chosenPlayer = null;
+	protected $mobsControl = null;
+
+	protected function initEntity(){
+		parent::initEntity();
+		$this->mobsControl = MobsControl::getInstance();
+		$this->mobsControl->register($this);
+		$health = $this->mobsControl->getHealth($this->getName());
+		$this->setMaxHealth($health);
+		$this->setHealth($health);
+	}
+	
+	public function kill() {
+		if ($this->getHealth() > 0) {
+			$name = $this->getName();
+			print "Monster:kill called for $name \n";
+			$this->mobsControl->deregister($this);
+		}
+		parent::kill();
+	}
+		
+	public function entityBaseTick($tickDiff = 1){
+		$hasUpdate = parent::entityBaseTick($tickDiff);
+		if($this->chunk === null or $this->closed){
+			print "Monster: No chunk or closed \n";
+			return;
+		}
+		if ($this->mobsControl == null) {
+			$this->mobsControl = MobsControl::getInstance();
+		}
+		$mobsState = $this->mobsControl->getState();
+		if ($mobsState == MobsControl::STATE_KILL) {
+			print "Kill command sent, so killing monster \n";
+			$this->kill();
+			print "Killed monster \n";
+			return;
+		} else if ($mobsState == MobsControl::STATE_SLEEP) {
+			return;
+		}
+		$speed = $this->mobsControl->getSpeed($this->getName());;
+		$attackDamage = $this->mobsControl->getAttackDamage($this->getName());
+		$withinProximity = $this->mobsControl->getProximity($this->getName());;
+// 		print "Got attack damage as: $attackDamage \n";
+		$player = $this->choosePlayer($withinProximity);
+		if ($player != null) {			
+			$proximity = $this->getPlayerProximity($player, $withinProximity);
+			if ($proximity == 0) {
+				$ev = new EntityDamageByEntityEvent($this, $player, EntityDamageEvent::CAUSE_ENTITY_ATTACK, $attackDamage);
+				$player->attack($attackDamage, $ev);
+			} else if ($proximity == 1) {
+				$monsterPosition = new PositionAndOrientation($this);
+				if ($this->x < $player->getX()) {
+					if ($this->z < $player->getZ()) {
+						$monsterPosition->tryNorthWest($speed);
+					} else if ($this->z > $player->getZ()) {
+						$monsterPosition->trySouthWest($speed);
+					} else {
+						$monsterPosition->tryWest($speed);
+					}
+				} else if ($this->x > $player->getX()) {
```

Comment:
For contributing guidelines: read CONTRIBUTING.md in the repo. You also see a link to it when creating a pull request.